src/models/tools/geometry/geometry-content.ts

```diff
@@ -189,8 +204,19 @@ export const GeometryContentModel = types
         removeObjects,
         updateObjects,
         createPolygonFromFreePoints,
+        findObjects,
         applyChange: _applyChange,
-        syncChange
+        syncChange,
+
+        suspendSync() {
+          ++suspendCount;
+        },
+        resumeSync() {
+          if (--suspendCount <= 0) {
```

Comment:
Ooh semaphores! Feels like another one of those "hopefully javascript is single-threaded enough for this to work" things.